changelog/unreleased/graceful-shutdown-server

```diff
@@ -0,0 +1,5 @@
+Bugfix: server is now shutdown cleanly on TERM or INT signals
+
+Server now listens for TERM and INT signals and cleanly closes down the http.Server and listener.
+
+This is particularly useful when listening on a unix socket, as the server will remove the socket file from it shuts down.
```

Comment:
Doesn't the OS already take care of that?

Reply:
No, at least not on the Ubuntu I tested on, and from a search it seems plenty of others have been surpised by similar. Explicity closing down the server, which I believe calls `.Close()` on the listener seems to do the right thing.

I'll split the PRs as requested, but I think it best to merge the shutdown handling before the unix socket one.